table/tables/tables.go

```diff
@@ -828,26 +984,54 @@ func (t *TableCommon) AddRecord(sctx sessionctx.Context, r []types.Datum, opts .
 			}
 			row = append(row, value)
 			colIDs = append(colIDs, col.ID)
+			// the ddl is modify-col and the col is write-only or write-reorg
+			if numChecksums > 1 {
```

Comment:
Is it possible to abstract the common checksum logic in `AddRecord` and `UpdateRecord` into a utility of `TableCommon`? Not quite sure it looks like there is something similar.